### PR TITLE
Add support for private endpoints and Async Operations

### PR DESCRIPTION
**Pull request by arolshan**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by arolshan** [comment]
Required swagger for Private Endpoints onboarding. 

See for reference (already approved swaggers):

[SQL RP implementation1](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateEndpointConnections.json)

[SQL RP implementation2](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/PrivateLinkResources.json)

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json [hunk 1]
```diff
@@ -990,6 +940,318 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Gets list of private endpoint connections on a workspace.": {
+            "$ref": "./examples/PrivateEndpointConnectionList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.OperationalInsights/locations/{location}/asyncOperations/{operationId}": {
```
Comment (ryansbenson) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: The OperationsStatus and OperationResult APIs are not required in swagger (though are in the RP manifest). 

That said, this API should be called 'operationsStatus' not 'asyncOperations' for consistency across all RPs.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json [hunk 2]
```diff
@@ -1469,6 +1722,185 @@
...
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
```
Comment (ryansbenson) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: Resource contains 'location' and I believe this is a proxy API. Should this be /definitions/ProxyResource?

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json [hunk 3]
```diff
@@ -1469,6 +1722,185 @@
...
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "PrivateLinkServiceConnectionStateProperty": {
+      "description": "State of the private endpoint connection.",
+      "type": "object",
+      "required": ["status", "description"],
+      "properties": {
+        "status": {
```
Comment (ryansbenson) [inline specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: Is status computed or an input? Is it an Enum?

Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: its input.

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json (hunk 2 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: youre right, thanks

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json [hunk 4]
```diff
@@ -990,6 +940,318 @@
...
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/privateLinkResources": {
+      "get": {
+        "tags": ["PrivateLinkResources"],
+        "description": "Gets the private link resources that need to be created for a Log Analytics Workspace.",
+        "operationId": "PrivateLinkResources_ListByDatabaseAccount",
+        "x-ms-pageable": {
+          "nextLinkName": null
```
[the remark this answers is not in the document]
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: sure no problem

File: specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json (hunk 1 above)
Comment (arolshan) [reply specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2015-11-01-preview/OperationalInsights.json]: okay changed to operationStatuses to be consisted with other RPs

**Comment by lirenhe** [comment]
@arolshan , could you take a look at the issues reported by Prettier?>

**Comment by arolshan** [comment]
It says to ignore the test. In any case it’s just noise, I did use prettier

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Review by ryansbenson — COMMENTED** [review]
Looks good from ARMs side. 

Regarding my comment around privateLinkResources: I'm assuming you resolved the comment because it is intentionally a readonly resource (no put/delete).

**Comment by lirenhe** [comment]
> It says to ignore the test. In any case it’s just noise, I did use prettier
> 
We may update that tag for PrettierCheck. Could you follow https://aka.ms/AA6h31t to fix the issue?

**Comment by arolshan** [comment]
> > It says to ignore the test. In any case it’s just noise, I did use prettier
> 
> We may update that description for PrettierCheck. Could you follow https://aka.ms/AA6h31t to fix the issue?

okay, I ran prettier.

**Review by lirenhe — APPROVED** [review]

**Pull request merged**